modules/test/playwright/pages/commerce/commerceProductAdminPage.ts

```diff
@@ -5,26 +5,92 @@
 
 import {Locator, Page} from '@playwright/test';
 
+import {ApplicationsMenuPage} from '../product-navigation-applications-menu/ApplicationsMenuPage';
+
 export class CommerceProductAdminPage {
+	readonly addProductRelationHeading: (
+		productName: string
+	) => Promise<Locator>;
+	readonly applicationsMenuPage: ApplicationsMenuPage;
 	readonly creationMenuNewButton: Locator;
+	readonly deleteMenuItem: Locator;
 	readonly generateSkusMenuItem: Locator;
+	readonly itemOptionMenu: Locator;
 	readonly managementToolbarSearchInput: Locator;
+	readonly modalAddButton: Locator;
+	readonly modalCancelButton: Locator;
 	readonly page: Page;
+	readonly productRelationsLink: Locator;
 	readonly productSkusLink: Locator;
+	readonly spareProductMenuButton: Locator;
+	readonly specificProductMenuLink: (productName: string) => Promise<Locator>;
+	readonly validProductCheckbox: (productName: string) => Promise<Locator>;
 
 	constructor(page: Page) {
-		this.creationMenuNewButton = page.getByLabel('New', {exact: true});
+		this.addProductRelationHeading = async (productName: string) => {
+			return page.getByRole('heading', {
+				name: 'Add New Product to ' + productName,
+			});
+		};
+		this.applicationsMenuPage = new ApplicationsMenuPage(page);
+		this.creationMenuNewButton = page
+			.getByTestId('creationMenuNewButton')
+			.nth(0);
+		this.deleteMenuItem = page.getByRole('menuitem', {name: 'Delete'});
 		this.generateSkusMenuItem = page.getByRole('menuitem', {
 			exact: true,
 			name: 'Generate All SKU Combinations',
 		});
+		this.itemOptionMenu = page.getByTestId('itemActionDropdown');
 		this.managementToolbarSearchInput = page
 			.getByTestId('management-toolbar')
 			.getByPlaceholder('Search', {exact: true});
+		this.modalAddButton = page.getByTestId('modalAddButton');
+		this.modalCancelButton = page.getByTestId('modalCancelButton');
+		this.page = page;
+		this.productRelationsLink = page.getByRole('link', {
+			exact: true,
+			name: 'Product Relations',
+		});
 		this.productSkusLink = page.getByRole('link', {
 			exact: true,
 			name: 'SKUs',
 		});
+		this.spareProductMenuButton = page.getByRole('menuitem', {
+			exact: true,
+			name: 'Add Spare Product',
+		});
+		this.specificProductMenuLink = async (productName: string) => {
+			return page.getByRole('link', {name: productName});
+		};
+		this.validProductCheckbox = async (productName: string) => {
+			return page
+				.frameLocator('#modalIframe')
+				.getByTestId('row')
+				.filter({hasText: productName})
+				.getByRole('checkbox', {disabled: false});
+		};
+	}
+
+	async addSpareProductRelation() {
+		await Promise.all([
+			this.goToProductRelations(),
+			this.page.waitForResponse(
+				(resp) =>
+					resp.status() === 200 &&
+					resp
+						.url()
+						.includes(
+							'screenNavigationCategoryKey=product-relations'
+						)
+			),
```

Comment:
I think this should go in the goToProductRelationships method, what do you think? Otherwise every time we need to use the method we need to remember to "wait for the response"